src/passa/internals/_pip.py

```diff
@@ -127,9 +131,37 @@ def _convert_hashes(values):
     return hashes
 
 
+def get_sdist(ireq, sources, hashes=None):
+    """Downloads an sdist and prepares it before handing off the built distribution with
+    pep517 metadata included (via a patched InstallRequirement object)
+    """
+
+    kwargs = _prepare_wheel_building_kwargs(ireq)
+    finder = _get_finder(sources)
+    ireq.populate_link(finder, False, False)
+    ireq.ensure_has_source_dir(kwargs["src_dir"])
+    egg_info_name = "%s.egg-info" % packaging.utils.canonicalize_name(ireq.name).replace("-", "_")
+    ireq._egg_info_dir = os.path.join(ireq.source_dir, egg_info_name)
+    ireq._egg_info_path = os.path.join(ireq.source_dir, egg_info_name)
```

Comment:
Some better heuristic would be needed to find egg-info directories not in the root directory. Passa for example has passa.egg-info inside `src`.

Reply:
Also if we already knows where `egg-ingo` is, its content can be read statically with `packaging.metadata.Metadata`, without realying on pip internals.

Reply:
Not always... we often need to run the setup script so it’s not just a static file. For the same reason, we don’t package `src/passa.egg-info` in the sdist. It’s generated when you install the source. I don’t think there’s any way around that, is there?

Reply:
The rule here is applied only to the ireq itself which calculates where it’s supposed to be installed. Wheels are easy to work with but I’m not too sure how easy these will be to install without using this construct. InstallRequirements are a pip construct we are relying on anyway. I guess we can try to avoid that, but there is no guarantee of compatibility.

Reply:
I experimented a bit, and unfortunately the dynamically built egg-info is still in `src`, not the source root. This is likely because pip still runs the same code in setuptools to generate it in the end. I’ve implemented some heuristic to detect it, will push a while later (doing some final experiments right now).

Also we really don’t need pip here, at least from my experiments, because we’re piggybacking this after `build_wheel`. That part already builds it.